sanskrit_parser/base/sanskrit_base.py

```diff
@@ -111,12 +130,25 @@ def getTags(self):
         return self.tags
 
     def __str__(self):
-        return self.transcoded(SLP1)
+        global denormalize
```

Comment:
I read through `__str__` scenario and `contextmanager` in https://github.com/kmadathil/sanskrit_parser/issues/56#issuecomment-351581640, but didn't fully understand.

Since all print statements use `SanskritObject.__str__`, denormalization should be done if the SanskritObject was formed with `strict_io=True`. Is this right understanding? Shouldn't we do a check like `if not self._strict_io: denormalize` in `SanskritObject.__str__`?

Is it possible to have a scenario where caller creates `s = SanskritObject(..., strict_io=False)` and still wants a denormalized output with `print(s)`?

Reply:
The object that ends up getting printed is often not the object that's initialized. For example, in the lexical analyzer, we start with a string, which becomes a SanskritObject, which then becomes a graph, which becomes paths, which become objects again.

Reply:
This makes sense now. Thanks!